instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -52,23 +52,31 @@ module Client
 
           def query(sql, options = {})
             attributes = client_attributes
-            case config[:db_statement]
-            when :include
-              attributes['db.statement'] = sql
-            when :obfuscate
-              attributes['db.statement'] = obfuscate_sql(sql)
-            end
+
             tracer.in_span(
               database_span_name(sql),
               attributes: attributes.merge!(OpenTelemetry::Instrumentation::Mysql2.attributes),
               kind: :client
-            ) do
+            ) do |span|
+              if OpenTelemetry::Trace.current_span.recording? && !OpenTelemetry::Instrumentation::Mysql2.attributes["db.statement"]
```

Comment:
So we don't `merge!` attributes twice. Probably could do this more cleanly.

Reply:
I kinda think it's not needed in general? In what circumstances would this situation actually happen? (Maybe it can, I'm just not sure when, where, or if it matters much)

Reply:
We need to be careful of a few things:
1. Overrides should always take precedence, including over the config.
2. As many attributes as possible should be supplied eagerly.
3. Hash merging and the implicit mutex access in `set_attribute` should be minimized.

Reply:
Co-sign what @fbogsany said. I think there's a reasonable case for explicitly overriding `db.statement` for applications that default to obfuscating or redacting mysql `db.statement` attributes but then explicitly override `db.statement` using `.attributes` when the know that the query is safe to include.

Reply:
I generally agree with what is said, but I will `:homer-bushes:` here because I think I'm lacking some context to really know what I'm talking about. 😆 

Never mind me, I was never here! 😄